includes/gateways/paypal/checkout-actions.php

```diff
@@ -154,32 +155,79 @@ function create_order( $purchase_data ) {
 			);
 		}
 
+		$order_subtotal = $purchase_data['subtotal'];
+		$items          = get_order_items( $purchase_data );
+		// Adjust the order subtotal if any items are discounted.
+		foreach ( $items as $item ) {
+			// A discount can be negative, so cast it to an absolute value for comparison.
+			if ( (float) abs( $item['discount'] ) > 0 ) {
+				$order_subtotal -= ( $item['discount'] * $item['quantity'] );
+			}
+
+			// The discount amount is not passed to PayPal as part of the $item.
+			unset( $item['discount'] );
+		}
+
+		$discount = 0;
+		// Fees which are not item specific need to be added to the PayPal data as order items.
+		if ( ! empty( $purchase_data['fees'] ) ) {
+			foreach ( $purchase_data['fees'] as $fee ) {
+				if ( ! empty( $fee['download_id'] ) ) {
+					continue;
+				}
+				// Positive fees.
+				if ( floatval( $fee['amount'] ) > 0 ) {
+					$items[]         = array(
+						'name'        => stripslashes_deep( html_entity_decode( wp_strip_all_tags( $fee['label'] ), ENT_COMPAT, 'UTF-8' ) ),
+						'unit_amount' => array(
+							'currency_code' => $currency,
+							'value'         => edd_sanitize_amount( $fee['amount'] ),
```

Comment:
Just in case, cast this to a string please. That's how PayPal wants all amount values.

```suggestion
							'value'         => (string) edd_sanitize_amount( $fee['amount'] ),
```